crates/chia-tools/src/bin/validate-blockchain-db.rs

```diff
@@ -183,6 +183,54 @@ features that are validated:
             let amount = u64::from_be_bytes(row.get::<_, [u8; 8]>(4).expect("missing amount"));
             additions.insert(coin_name, (ph, parent, amount, reward));
         }
+
+        // first ensure that the reward coins for this block are all included in
+        // the coin record table.
+        let rewards = block.get_included_reward_coins();
+        for add in &rewards {
+            let new_coin_id = add.coin_id();
+            let Some((ph, _parent, amount, coin_base)) = additions.get(new_coin_id.as_slice())
```

Comment:
```suggestion
            let Some(&(ph, _parent, amount, coin_base)) = additions.get(new_coin_id.as_slice())
```
Leveraging the reference we get from `get` allow us to compare `amount` and `coin_base` values directly.